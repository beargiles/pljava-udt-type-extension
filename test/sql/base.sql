\set ECHO 0
BEGIN;
\i sql/pljava-udt-type.sql
\set ECHO all

-- You should write your tests

SELECT pljava-udt-type('foo', 'bar');

SELECT 'foo' #? 'bar' AS arrowop;

CREATE TABLE ab (
    a_field pljava-udt-type
);

INSERT INTO ab VALUES('foo' #? 'bar');
SELECT (a_field).a, (a_field).b FROM ab;

SELECT (pljava-udt-type('foo', 'bar')).a;
SELECT (pljava-udt-type('foo', 'bar')).b;

SELECT ('foo' #? 'bar').a;
SELECT ('foo' #? 'bar').b;

ROLLBACK;
