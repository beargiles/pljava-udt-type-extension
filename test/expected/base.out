\set ECHO 0
-- You should write your tests
SELECT pljava-udt-type('foo', 'bar');
 pljava-udt-type 
-----------------
 (foo,bar)
(1 row)

SELECT 'foo' #? 'bar' AS arrowop;
  arrowop  
-----------
 (foo,bar)
(1 row)

CREATE TABLE ab (
    a_field pljava-udt-type
);
INSERT INTO ab VALUES('foo' #? 'bar');
SELECT (a_field).a, (a_field).b FROM ab;
  a  |  b  
-----+-----
 foo | bar
(1 row)

SELECT (pljava-udt-type('foo', 'bar')).a;
  a  
-----
 foo
(1 row)

SELECT (pljava-udt-type('foo', 'bar')).b;
  b  
-----
 bar
(1 row)

SELECT ('foo' #? 'bar').a;
  a  
-----
 foo
(1 row)

SELECT ('foo' #? 'bar').b;
  b  
-----
 bar
(1 row)

ROLLBACK;
